### PR TITLE
Make eslint work on static ts files

### PR DESCRIPTION
**Pull request by RubenRBS**
Opening this PR as I don't know if this is the best way we have to express the idea. But it's working

### CODE REVIEW

File: static/.eslint-ce-static.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  - jsdoc
+  - sonarjs
+  - unicorn
+extends:
+  - ../.eslint-license-header.yml
+  - eslint:recommended
+env:
+  browser: true
+  node: true
+  es6: false
```
Comment (matsjla) [inline static/.eslint-ce-static.yml:12]: I suppose we could add a comment here to enable es6 once all the old es5 js code is gone?

File: static/.eslint-ce-static.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  jsdoc/valid-types: warn
+  sonarjs/no-collection-size-mischeck: error
+  sonarjs/no-redundant-boolean: error
+  sonarjs/no-unused-collection: error
+  sonarjs/prefer-immediate-return: error
+  sonarjs/prefer-object-literal: error
+  sonarjs/prefer-single-boolean-return: error
+  unicorn/filename-case: error
+parserOptions:
+  ecmaVersion: 5
```
Comment (matsjla) [inline static/.eslint-ce-static.yml:82]: Same goes for this

**Review by matsjla — COMMENTED** [review]
I'm certainly not an eslint config wizard, but it looks alright to me

File: test/filter-tests.js [hunk 3]
```diff
@@ -80,7 +80,7 @@ function testFilter(filename, suffix, filters) {
     The before() hooks on mocha are for it()s - They don't execute before the describes!
     That's sad because then we can't have cases be loaded in a before() for every describe child to see.
  */
-describe('Filter test cases', function () {
+describe.skip('Filter test cases', function () {
```
Comment (RubenRBS) [inline test/filter-tests.js:83]:
```suggestion
describe('Filter test cases', function () {
```

File: static/.eslint-ce-static.yml (hunk 1 above)
Comment (partouf) [reply static/.eslint-ce-static.yml:12]: As long as webpack does magic to keep it es5, a couple more browsers will otherwise break.

Comment (mattgodbolt) [reply static/.eslint-ce-static.yml:12]: Agreed: or at least the option (I suspect ES5 is already broken by Monaco et al...)

**Review by mattgodbolt — APPROVED** [review]
Looks good to me! GH is mentioning tons are things that are problematic (in its "beta" of "Unchecked files with check annotations" - I don't know if that's related or not?)

**Comment by RubenRBS** [comment]
I think all should be good now. We might want to hurry merging this one as it touches lots of files so that it's not a merging hell

**Review by partouf — APPROVED** [review]

**Pull request merged**